### ISSUE TITLE
[TODO] Complete the Dashboard component 

### ISSUE THREAD

**Issue by gbowne1** [gbowne1/reactsocialnetwork#83]
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

Using Material UI etc 

Complete the dashboard for ome of the following
- User Dashboard (General items not in settings)
- Admin & Moderator dashboard (for chat and posts)
- Analytics Dashboard
- Groups & Events dashboard (Creating and managing groups and events)

Could be a tabbed view controlled by buttons or breadcrumbs or.. 

**Describe alternatives you've considered**
I have considered using this as several things
- Analytics dashboard
- Admin / Moderator dashboard
- User dashboard for general thing

Ideally we would have all 3.  whether that be separate components or available via a nav inside the component??

**Additional context**

**Comment by gbowne1** [comment]
Ideas @pawel975 ?

**Comment by pawel975** [comment]
I think Admin & Moderator dashboard could be in form of modal popup which can be opened by clicking on any user post to handle moderation etc. 

If it goes about user dashboard not sure what can be included there. I think all needed things could be gathered in user profile settings like, writing description, including profile picture etc. 

Analytics could be separate admin panel.

**Comment by gbowne1** [comment]
@pawel975 I totally agree on this.with a few comments.

Users Analytics can be seperate from Admin Analytics

We havent implemented userland stuff much yet so yeah.

User actually would have a lot of controls and settings. 

Analytics overall can be a big thing for both staff and users.

I would like to give admin and mod rights to all posts and comments etc. within  Controls for banning or warning users with a heavy emphasis on warnings.  

Show me a first iteration of this.

The user controls for Groups on Facebook and Nextdoor are terrible.  This was my main reason for starting this project.

**Comment by gbowne1** [comment]
My thoughts for Dashboard
![ReactSocialTemplate-Dashboard](https://user-images.githubusercontent.com/47549872/224817213-b393ddcf-2c68-468d-9b27-550ccf5cf4d6.jpg)

**Comment by goutham03062001** [comment]
Hey @gbowne1. I would like to work upon the design of Dashboard component. 

Let me know what you want to including in this component. I've gone through the issue description you have provided on top.

Here are my thoughts for this Dashboard component.

**1) USER DASHBOARD**

In this component, we can show his posts, friends, followers,following and profile followed by bio. And we can provide him full control of this dashboard.. 

**2) ANALYTICS DASHBOARD**

My thoughts for this component is to view the activity of users based on the location. We can gathere what kind of meetings they are attending and what posts they are liking or interested in. And what are the age groups of people mostly are from that region. And how much time he is spending time on our website. And there are lot to do with their data.

**3) ADMIN DASHBOARD**

Well, this is our main component in the entire application and for sure it must be very secure. We can include all users data and we can take control of every user, and we can include what kind of content they are viewing, and also we can get better ideas by working on this in the future.

**Comment by gbowne1** [comment]
Screenshot for context:

![Screenshot 2023-04-19 at 21-51-22 React Social Media App](https://user-images.githubusercontent.com/47549872/233261476-479dcf42-214c-4d63-8318-0dc71c260506.png)

The actual component style we use across all components:

![Screenshot 2023-04-19 at 21-52-59 React Social Media App](https://user-images.githubusercontent.com/47549872/233261769-b5190069-0796-4261-8613-22384021904c.png)

This sounds great to me.  This page is in /src/pages/Dashboard with a css and a jsx file. If you need to seperate the 3 into components, I would probably even just copy the code from Dashboard.jsx and Dashboard.css and reuse it to make the components you need to save rework on componentry and just so everyones on the same page.

Our css is typically defined like .Component-section { ... };

MUI 5.11 includes a lot of great icons.  Like I showed in the post above I made on March 13.. you could build the different sections within the Dashboard.jsx component.  

Components typically go into /src/pages/{Componentname/{Componentname}.jsx  

I think we need to make a boilerplate component people can reuse for our component panel(s).  (there will be a few)

Ideas? @pawel975 @manuel

**Comment by gbowne1** [comment]
Yeah probably easier to see the Dashboard.jsx compnent in dark mode..

![Screenshot 2023-04-19 at 22-34-32 React Social Media App](https://user-images.githubusercontent.com/47549872/233267747-df2a5f0d-e83c-418b-94a9-868fa1ef6ec2.png)

**Comment by goutham03062001** [comment]
Hey I worked on designing user dashboard. 

Here is the screenshot what I've done so far.

![image](https://user-images.githubusercontent.com/66135872/233686989-0a48b089-675c-4291-9dca-b0392dc8eb4f.png)

You can now take a look at what I've done.

https://www.figma.com/file/8G8ZfMDBQPBbskNwJo74O6/Untitled?node-id=0%3A1&t=DNQ86EchUmhXc47g-1

**Comment by pawel975** [comment]
> Hey I worked on designing user dashboard.
> 
> Here is the screenshot what I've done so far.
> 
> ![image](https://user-images.githubusercontent.com/66135872/233686989-0a48b089-675c-4291-9dca-b0392dc8eb4f.png)
> 
> You can now take a look at what I've done.
> 
> https://www.figma.com/file/8G8ZfMDBQPBbskNwJo74O6/Untitled?node-id=0%3A1&t=DNQ86EchUmhXc47g-1

Looks good but be sure to follow colors, font's etc from /src/variables.css

**Comment by gbowne1** [comment]
@pawel975 yes.  It's just a design though.  

@goutham03062001 We already have a theme switch that controls the entire theme.  There's no need for one in the User Dashboard.  We *could* do some userland settings in a display/settings page, outside of the user icon / avatar settings page.  Thats a nice design for a user dashboard though, but will need some internal componentry.  I am on board with everything there.

**Comment by gbowne1** [comment]
@goutham03062001 are you still working on this?

**Comment by goutham03062001** [comment]
Nope.

**Comment by gbowne1** [comment]
Ok thanks

**Comment by gbowne1** [comment]
To see the actual dashboard to use in your development go to:

http://localhost:3000/dashboard 

the files you will need to modify are:

client/src/pages/Dashboard.css 
client/src/pages/Dashboard.jsx
